util/packaging/docker/README.md

```diff
@@ -0,0 +1,145 @@
+
+### [`chapel/chapel-gasnet:<version>`](https://hub.docker.com/r/chapel/chapel-gasnet/)
+* `1.27.0`, `latest`
+* `1.26.0`
+* `1.25.1`
+* `1.25.0`
+
+The Chapel core image (above), rebuilt with `CHPL_COMM=gasnet` and
+`GASNET_SPAWNFN=L`. Simulates a multilocale Chapel platform within the Docker
+container.
+
+Multilocale Chapel brings additional requirements, unrelated to Docker. The
+`chpl` compilation produces two binary files (e.g. `hello_real` as well as
+`hello`). When you run the binary, you need another command line parameter,
+`-nl #`, to specify the number of locales. Please see
+[Multilocale Chapel Execution](https://chapel-lang.org/docs/usingchapel/multilocale.html)
+for details.
+
+
+## How to use the image
+
+### Compile your Chapel program
+
+The image can be used to compile and run your Chapel program inside the
+container.
+
+In the following example, you start a shell in the container and use the
+interactive session to compile and run your Chapel program, all within the
+container. The `-v` argument mounts the current directory in the container,
+`-w` sets the working directory to this mounted directory, `-i` gives you STDIN,
+and `-t` allocates a psuedo-TTY, giving you STDOUT and STDERR.
+
+The Chapel compiler, `chpl`, is invoked much like `gcc`, with argument `-o` to
+specify the name of the output binary. Then the binary file, `hello`, is run in
+the usual way.
+
+```
+$ echo 'writeln("Hello, world!");' > hello.chpl
+
+$ docker run --rm -it -v "$PWD":/myapp -w /myapp chapel/chapel /bin/bash
+root@xxxxxxxxx:/myapp# chpl -o hello hello.chpl
+root@xxxxxxxxx:/myapp# ./hello
+Hello, world!
+```
+&nbsp;
+
+Alternatively, you could compile your program in one instance of the container,
+and run it in another, for the same end result.
+```
+$ docker run --rm -v "$PWD":/myapp -w /myapp chapel/chapel chpl -o hello hello.chpl
+
+$ docker run --rm -v "$PWD":/myapp -w /myapp chapel/chapel ./hello
+Hello, world!
+```
+&nbsp;
+
+On 64-bit Linux systems, the resultant binary can sometimes be run outside the
+container, if it was compiled with the `--static` option:
+```
+$ docker run --rm -v "$PWD":/myapp -w /myapp chapel/chapel chpl --static -o hello hello.chpl
+
+$ ./hello
+Hello, world!
+```
+&nbsp;
+
+### Develop chapel with Docker
+
+It is possible to do development work on Chapel itself with Docker; the below
+workflow example performs editing outside of Docker, followed by compilation and
+execution within Docker.
+
+```
+$ cd [Chapel source tree root]                  # acquire Chapel sources
+$ vim [Chapel source files]                     # hack on Chapel code
+$ docker build -t chapel-dev .                  # build a new image locally from the modified code
+$ docker run --rm -it chapel-dev make test      # test your modified code as desired
+$ [...]                                         # repeat edit-compile-run cycle as desired...
+```
+
+If you have a fast computer you can speed up the building step by modifying the
```

Comment:
Do you just mean change the command above on line 123?

Reply:
I mean actually modifying the invocation of `make` in the Dockerfile (which is run when we `docker build` from the Dockerfile on line 122). A developer working on a computer that could handle multiple build threads might like to modify that to get significantly faster compiles in the edit-compile-run cycle.

The Dockerfile uses `make` without the `-j` flag because:
- It's the lowest common denominator for building on any system, which potentially may be single-core.
- Clear output from the build is the priority over speed in all Docker build situations except this development-cycle case.